base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1168,20 +1207,61 @@ pub unsafe extern "C" fn encrypted_value_encrypt(
 /// memory leak
 #[no_mangle]
 pub unsafe extern "C" fn encrypted_value_decrypt(
+    wallet: *mut TariWallet,
```

Comment:
Same here, this method can be used to arbitrarily decrypt some data

Reply:
I would argue that we might keep this method. It is very handy for a wallet to be able to detect with a high degree of certainty which UTXOs it owns. That being said, a strategy of detect-ownership-then-request-all-information may very well be completely implemented in the base layer wallet and never touch the front-end.

W.r.t. "_arbitrarily decrypt some data_"; I think it is equally true or equally difficult for any client and not only clients using the FFI interface.